src/test/java/org/opensearch/security/dlic/dlsfls/RenameFieldResponseProcessorTest.java

```diff
@@ -0,0 +1,228 @@
+                        + "\"FlightNum\": \"X98CCZO\","
+                        + "\"DestAirportID\": \"VE05\","
+                        + "\"Dest\": \"Venice Marco Polo Airport\","
+                        + "\"DestCountry\": \"IT\","
+                        + "\"OriginAirportID\": \"CPT\","
+                        + "\"Origin\": \"Cape Town International Airport\","
+                        + "\"OriginCountry\": \"ZA\","
+                        + "\"FlightDelay\" : false,"
+                        + "\"Canceled\": false"
+                        + "}",
+                    XContentType.JSON
+                )
+        ).actionGet();
+        tc.index(
+            new IndexRequest("flights").id("2")
+                .setRefreshPolicy(RefreshPolicy.IMMEDIATE)
+                .source(
+                    "{"
+                        + "\"FlightNum\": \"UFK2WIZ\","
+                        + "\"DestAirportID\": \"SYD\","
+                        + "\"Dest\": \"Venice Marco Polo Airport\","
+                        + "\"DestCountry\": \"IT\","
+                        + "\"OriginAirportID\": \"FRA\","
+                        + "\"Origin\": \"Venice Marco Polo Airport\","
+                        + "\"OriginCountry\": \"IT\","
+                        + "\"FlightDelay\" : false,"
+                        + "\"Canceled\": true"
+                        + "}",
+                    XContentType.JSON
+                )
+        ).actionGet();
+    }
+
+    @Test
+    public void testMaskedField() throws Exception {
+        setup(new DynamicSecurityConfig().setSecurityRoles(ROLES_FILE).setSecurityRolesMapping(ROLES_MAPPINGS_FILE));
+
+        HttpResponse res;
+        res = rh.executePostRequest("/flights/_search", emptyQuery, asUserA);
+        assertThat(res.getBody(), not(containsString("\"FlightNum\": \"9HY9SWR\",")));
+        System.out.println(res);
+        String testRenameMaskedFieldPipeline = "{"
+            + "\"description\": \"A pipeline to rename masked field 'FlightNum' to 'FlightNumNew'\","
+            + "\"response_processors\": ["
+            + "{"
+            + "\"rename_field\": {"
+            + "\"target_field\": \"FlightNumNew\","
+            + "\"field\": \"FlightNum\""
+            + "}"
+            + "}"
+            + "]"
+            + "}";
+        res = rh.executePutRequest("/_search/pipeline/test-rename-masked-field", testRenameMaskedFieldPipeline, asAdmin);
+        assertThat(res.getStatusCode(), equalTo(HttpStatus.SC_OK));
+
+        // Search with this pipeline should succeed and the value of the new field "FlightNumNew" should also be masked
+        res = rh.executePostRequest("/flights/_search?search_pipeline=test-rename-masked-field&size=1", emptyQuery, asUserA);
+        assertThat(res.getStatusCode(), equalTo(HttpStatus.SC_OK));
+        assertThat(res.getBody(), containsString("\"FlightNumNew\":"));
+        assertThat(res.getBody(), not(containsString("\"FlightNumNew\": \"9HY9SWR\",")));
```

Comment:
Nice! Took me a bit to get it working, but since the fields are nested inside a document's `_source`, I needed to specify the dot path. In this case, I changed it to `res.findValueInJson("hits.hits[0]._source.FlightNumNew")`.